### PR TITLE
Remove humanoid_league_misc and add caching to tts

### PR DESCRIPTION
**Pull request by Flova**
# Summary

This pr moves the last two remaining humanoid league packages out of humanoid_league_mics. They are also renamed accordingly. The team comm is renamed to `bitbots_team_communication` and the speaker is renamed to `bitbots_tts`, which is a more fitting name imo.. The team comm will also be migrated further into ros sports in the near future, so this is only an intermediate solution. 

In addition to that caching is added to the tts (as proposed in #228). The outputs are generated at the first call, but persistently stored afterward. I like this approach as it still is quite flexible, while increasing performance overall and being dead simple (supported out of the box by the tts we use).

## Proposed changes
- Move folder and change references in code
- Set cache dir flag in mimic server
- Cleanup references to old humanoid league packages in docs

### CODE REVIEW

**Comment by Flova** [comment]
Testing of the team comm was not successful, but it could be my testing procedure.

**Comment by jaagut** [comment]
This does not fully resolve #228, as I would like to improve the spoken text (see issue for that).

File: bitbots_motion/bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/pickup.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from humanoid_league_speaker.speaker import speak
+from bitbots_tts.tts import speak
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_motion/bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/pickup.py:1]: I don't really know. Maybe it is a bit shorter. But I didn't want to change too much functionality.

File: bitbots_motion/bitbots_hcm/bitbots_hcm/humanoid_control_module.py [hunk 2]
```diff
@@ -5,11 +5,11 @@
 
 import rclpy
 from ament_index_python import get_package_share_directory
+from bitbots_tts.tts import speak
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_motion/bitbots_hcm/bitbots_hcm/humanoid_control_module.py:8]: I also think the speak function is just a publisher to the speak topic

**Comment by Flova** [comment]
> This does not fully resolve #228, as I would like to improve the spoken text (see issue for that).

Ah you are right nvm

**Review by jaagut — APPROVED** [review]

**Comment by Flova** [comment]
> Testing of the team comm was not successful, but it could be my testing procedure.

Is this resolved?

**Pull request merged**